cmd/relay-gossip/client/client.go

```diff
@@ -115,6 +129,52 @@ func (c *Client) Sub(ch chan drand.PublicRandResponse) UnsubFunc {
 	}
 }
 
+func (c *Client) Watch(ctx context.Context) <-chan client.Result {
+	innerCh := make(chan drand.PublicRandResponse)
+	outerCh := make(chan dclient.Result)
+	end := c.Sub(innerCh)
+
+	go func() {
+		for {
+			select {
+			case resp, ok := <-innerCh:
+				if !ok {
+					close(outerCh)
+					return
+				}
+				outerCh <- &result{resp.Round, resp.Randomness, resp.Signature}
+			case <-ctx.Done():
+				close(outerCh)
+				end()
+				// drain leftover on innerCh
+				for range innerCh {
```

Comment:
The sender won't react to the ctx cancellation right away. It will try sending more to the channel and it will lock forever if the channel is not being drained.